### ISSUE TITLE
py-compile: Missing argument to --destdir.

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#1238]
As discussed in #627, `make install` sometimes fails in the python bindings with
```
$ cd src/bindings/python
$ sudo make install
Making install in flux
make[1]: Entering directory '/home/garlick/proj/flux-core/src/bindings/python/flux'
Making install in core
make[2]: Entering directory '/home/garlick/proj/flux-core/src/bindings/python/flux/core'
echo Copying core libraries to where they can be used by python out of tree
Copying core libraries to where they can be used by python out of tree
[ "../../../../.." != "../../../../.." ] && cp  ../../../../../src/bindings/python/flux/core/*.py ./ || true
make[3]: Entering directory '/home/garlick/proj/flux-core/src/bindings/python/flux/core'
make[3]: Nothing to be done for 'install-exec-am'.
 /bin/mkdir -p '/usr/local/lib/python2.7/dist-packages/flux/core'
 /usr/bin/install -c -m 644 __init__.py watchers.py inner.py handle.py trampoline.py '/usr/local/lib/python2.7/dist-packages/flux/core'
../../../../../config/py-compile: Missing argument to --destdir.
Makefile:448: recipe for target 'install-fluxpycorePYTHON' failed
make[3]: *** [install-fluxpycorePYTHON] Error 1
make[3]: Leaving directory '/home/garlick/proj/flux-core/src/bindings/python/flux/core'
Makefile:546: recipe for target 'install-am' failed
make[2]: *** [install-am] Error 2
make[2]: Leaving directory '/home/garlick/proj/flux-core/src/bindings/python/flux/core'
Makefile:747: recipe for target 'install-recursive' failed
make[1]: *** [install-recursive] Error 1
make[1]: Leaving directory '/home/garlick/proj/flux-core/src/bindings/python/flux'
Makefile:475: recipe for target 'install-recursive' failed
make: *** [install-recursive] Error 1
```

This can be worked around by setting `DESTDIR=/`  on the command line or installing from a clean checkout.

We should probably address this in Makefile.am by making sure `py-compile` is never called with an empty `--destdir` optarg to avoid future confusion.

**Comment by garlick** [comment]
py-compile and the makefile rules that call it are all generated by automake.

I can reproduce the problem in  `automake 1.15-4ubuntu1`

There _seems_ to be a fix in `automake 1.11.2`:

```
commit dd8b570a6a55cb20690cdc2c312309338d2495d3
Author: Stefano Lattarini <stefano.lattarini@gmail.com>
Date:   Wed Jun 8 14:55:01 2011 +0200

    py-compile: the '--destdir' option now accepts a blank argument
    
    * lib/py-compile (Option parsing): Do not count an empty argument
    to `--destdir' or `--basedir' as a missing argument.
    * lib/python.am: Simplify accordingly, passing the `--destdir'
    option to py-compile unconditionally, even if `$(DESTDIR)' is
    empty.
    * NEWS: Update.
```
From NEWS entry for 1.11.2:
```
  - The py-compile script now accepts empty arguments passed to the options
    `--destdir' and `--basedir', and complains about unrecognized options.
    Moreover, a non-option argument or a special `--' argument terminates
    the list of options.
```

Hmm, maybe there was a regression or maybe this wasn't the fix.

**Comment by grondo** [comment]
Apparently no longer an issue.
